libraries/base/Data/Semigroup.hs

```diff
@@ -18,21 +18,36 @@
 -- Maintainer  :  libraries@haskell.org
 -- Stability   :  provisional
 -- Portability :  portable
+-- 
+-- A type @a@ is a 'Semigroup' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one. 
 --
--- In mathematics, a semigroup is an algebraic structure consisting of a
--- set together with an associative binary operation. A semigroup
--- generalizes a monoid in that there might not exist an identity
--- element. It also (originally) generalized a group (a monoid with all
--- inverses) to a type where every element did not have to have an inverse,
--- thus the name semigroup.
+-- ==== __Examples__
+-- >>> sconcat (1 :| [2, 3, 4]) :: Min Int
```

Comment:
As suggested in #177, it wouldn't hurt to first briefly introduce the structures being used here. Currently there is a bit of assumed familiarity that a new-comer may not possess.